### PR TITLE
Move watch/versioned to pkg/apis/meta/v1/watch.go

### PR DESCRIPTION
**Pull request by smarterclayton**
Watch is a part of the server API

### CODE REVIEW

**Comment by smarterclayton** [comment]
@kubernetes/sig-api-machinery-misc ongoing move of things into meta/v1, reviewer appreciated (mostly mechanical).

Note that OpenAPI generation is disabled for the watch Event to avoid #38071

File: pkg/client/restclient/watch/encoder.go [hunk 1]
```diff
@@ -27,6 +28,7 @@ import (
 // Encoder serializes watch.Events into io.Writer. The internal objects
 // are encoded using embedded encoder, and the outer Event is serialized
 // using encoder.
+// TODO: this type is only used by tests
```
Comment (sttts) [inline pkg/client/restclient/watch/encoder.go]: what's the todo then? move?

File: pkg/apis/meta/v1/watch.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 // Event represents a single event to a watched resource.
 //
 // +protobuf=true
-// +k8s:openapi-gen=true
+// +k8s:openapi-gen=false
```
Comment (sttts) [inline pkg/apis/meta/v1/watch.go]: intentional?

File: pkg/apis/meta/v1/watch.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package versioned
+package v1
```
Comment (sttts) [inline pkg/apis/meta/v1/watch.go]: A bit strange now to have `InternalEvent` in `v1` here and not one level up.

**Comment by sttts** [comment]
In general looks good. Only oddity: `InternalEvent` should be in `pkg/apis/meta` and called `WatchEvent` as well.

**Comment by smarterclayton** [comment]
InternalEvent is special - it exists only as a placeholder, and is
representative of the "internal type is the same as external type"
abstraction we create.  Basically, it needs to stay as is.

File: pkg/client/restclient/watch/encoder.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/client/restclient/watch/encoder.go]: Probably to figure out whether we just remove it.

File: pkg/apis/meta/v1/watch.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/apis/meta/v1/watch.go]: Yes, but then I ended up renaming the type because swagger was broken anyway.

**Comment by smarterclayton** [comment]
InternalEvent really is just a shim to have two types in the code that the Scheme can tie to a Kind, but to not actually have two Go types (and thus no conversion is necessary).

**Comment by sttts** [comment]
Got it. lgtm.

**Comment by smarterclayton** [comment]
@k8s-bot unit test this

**Comment by smarterclayton** [comment]
@k8s-bot node e2e test this

**Comment by smarterclayton** [comment]
Tests are green, trap is clean.  Labelling.

**Comment by sttts** [comment]
@k8s-bot node e2e test this

**Comment by sttts** [comment]
@k8s-bot cri node e2e test this

**Comment by ncdc** [comment]
FYI @k8s-oncall @eparis I think this broke `hack/verify-godeps.sh`:

```
I0110 03:08:47.562] Verifying hack/make-rules/../../hack/verify-godeps.sh
W0110 03:08:54.699] Note: checking out 'v74'.
W0110 03:08:54.700] 
W0110 03:08:54.701] You are in 'detached HEAD' state. You can look around, make experimental
W0110 03:08:54.701] changes and commit them, and you can discard any commits you make in this
W0110 03:08:54.702] state without impacting any branches by performing another checkout.
W0110 03:08:54.703] 
W0110 03:08:54.704] If you want to create a new branch to retain commits you create, you may
W0110 03:08:54.705] do so (now or later) by using -b with the checkout command again. Example:
W0110 03:08:54.705] 
W0110 03:08:54.706]   git checkout -b new_branch_name
W0110 03:08:54.706] 
W0110 03:08:54.707] HEAD is now at f575518... Fix the silly file
I0110 03:08:57.405] godep v74 (linux/amd64/go1.7.4)
I0110 03:08:57.406] Starting to download all kubernetes godeps. This takes a while
I0110 03:11:36.121] Download finished
I0110 03:11:36.403] godep v74 (linux/amd64/go1.7.4)
W0110 03:16:03.112] godep: Package (k8s.io/kubernetes/pkg/watch/versioned) not found
I0110 03:16:03.232] Removing /tmp/gopath.8f81px
I0110 03:16:06.620] [0;31mFAILED[0m   hack/make-rules/../../hack/verify-godeps.sh	439s
```

I'm seeing if I can come up with a quick fix

**Comment by sttts** [comment]
Blocked by this as well. @ncdc if you need any support, ping me on irc.

**Comment by ncdc** [comment]
@sttts we could race :-) I can probably work on this for another 30-45 minutes before I have to step away for a bit

**Pull request merged**